crates/cext/src/sparse_observable.rs

```diff
@@ -0,0 +1,749 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::exit_codes::{CInputError, ExitCode};
+use num_complex::Complex64;
+use qiskit_accelerate::sparse_observable::{BitTerm, SparseObservable, SparseTermView};
+
+/// @ingroup QkSparseTerm
+/// A term in a [SparseObservable].
+///
+/// This term does not own the data, but only contains pointers (and a size) to the
+/// bit terms, indices and the coefficient.
+#[repr(C)]
+pub struct CSparseTerm {
+    coeff: *mut Complex64,
```

Comment:
The trouble with `Complex64` in interfaces is only in the calling conventions of functions - we can't use a raw `Complex64` in them because the conventions of Rust structs aren't necessarily the same as for C's `double complex` on all platforms and compilers.  I think it's fine to have `coeff: Complex64` inline here, because it's memory-layout compatible with `double complex`, and that'll simplify creating one of these.

Reply:
Updated to use the value, not the pointer, in https://github.com/Qiskit/qiskit/commit/bc376c6f27b20c932c7eb4e3da7e63ffb9cbf2ea